src/Microsoft.Azure.WebJobs.ServiceBus/Listeners/ServiceBusListener.cs

```diff
@@ -51,11 +54,34 @@ private Task StartAsyncCore(CancellationToken cancellationToken)
             cancellationToken.ThrowIfCancellationRequested();
 
             _receiver = _messagingProvider.CreateMessageReceiver(_messagingFactory, _entityPath);
-            _receiver.OnMessageAsync(ProcessMessageAsync, _messageProcessor.MessageOptions);
+            var cloned = CloneMessageOptions(_messageProcessor.MessageOptions);
+            cloned.ExceptionReceived += (sender, args) => _handler.Capture(args.Exception);
+            _receiver.OnMessageAsync(ProcessMessageAsync, cloned);
 
             return Task.FromResult(0);
         }
 
+        internal static OnMessageOptions CloneMessageOptions(OnMessageOptions original)
+        {
+            var clone = new OnMessageOptions()
+            {
+                AutoComplete = original.AutoComplete,
+                AutoRenewTimeout = original.AutoRenewTimeout,
+                MaxConcurrentCalls = original.MaxConcurrentCalls
+            };
+            var field = typeof(OnMessageOptions).GetField("ExceptionReceived", BindingFlags.Instance | BindingFlags.NonPublic);
```

Comment:
Oy. It doesn't feel right doing reflection to clone this. I'm not sure if there's a better alternative, though. @MikeStall -- any recommendations on this?

Just to reiterate -- we're doing this because without it, every ServiceBusTrigger will wire up a new ExceptionReceived handler -- so any exception will get broadcast to every trigger. Is that right?